Gymgenius/Gymgenius/Controllers/ExerciseToProgramController.cs

```diff
@@ -0,0 +1,86 @@
+﻿using Gymgenius.bll;
+using Gymgenius.bo;
+using Gymgenius.dal;
+using GymGenius.BLL;
+using GymGenius.BO;
+using GymGenius.DAL;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.SignalR;
+
+namespace Gymgenius.Controllers
+{
+    [ApiController]
+    [Route("[controller]")]
+    public class ExerciseToProgram : ControllerBase
+    {
+        private readonly ExerciseToProgramManagment exerciseToProgramManagment;
+        private readonly ExerciseManagment exerciseManagment;
+        private readonly TrainingProgramManagment trainingManagment;
+
+        public ExerciseToProgram(IExerciseToProgramRepository exerciseToProgramDAL, IExerciseRepository exerciseDAL, ITrainingProgramRepository trainingProgramDAL)
+        {
+            exerciseToProgramManagment = new ExerciseToProgramManagment(exerciseToProgramDAL);
+            exerciseManagment = new ExerciseManagment(exerciseDAL);
+            trainingManagment = new TrainingProgramManagment(trainingProgramDAL);
+        }
+
+        [HttpGet("list_exercises/{id}")]
+        public ActionResult<IEnumerable<Exercise>> GetAllExerciseOfProgram(int id)
+        {
+            if (!trainingManagment.IsTrainingProgramExists(id))
+            {
+                return NotFound("No program found.");
+            }
+
+            return exerciseToProgramManagment.GetAllExerciseOfProgram(trainingManagment.GetTrainingProgramById(id));
+        }
+
+        [HttpGet("add_exercise_to_program/{exercise_name}/{program_id}")]
+        public ActionResult AddExerciseToProgram(string exercise_name, int program_id)
+        {
+            if (!exerciseManagment.IsExerciseExists(exercise_name))

```

Comment:
this logic should be in Training management, api controller just query the  data, not make any unnecessary logic